test_util/marathon.py

```diff
@@ -200,11 +200,46 @@ def _pool_for_marathon_app(app_id):
                 return None
 
         try:
-            return _pool_for_marathon_app(app_definition['id'])
+            return _poll_marathon_for_app_deployment(app_definition['id'])
         except retrying.RetryError:
             raise Exception("Application deployment failed - operation was not "
                             "completed in {} seconds.".format(timeout))
 
+    def ensure_deployments_complete(self, timeout=600):
+        """
+        This method ensures that, there are no pending deployments
+
+        :return: True if all deployments are completed within time out. Raises an exception otherwise.
+        """
+
+        @retrying.retry(wait_fixed=1000, stop_max_delay=120 * 1000, retry_on_exception=lambda x: False)
+        def _get_deployments_json():
```

Comment:
Why the retry here? We should be able to get deployments reliably, for the same reason we POST apps without retrying.